### PR TITLE
Add support for keyboard keys 1-10, and spacebar for Item Box Actions

### PR DESCRIPTION
**Pull request by alycii**
## What has been done?

When producing an item box (fletching logs, spinning wool, smelting bars, etc..) you can now press keys on your keyboard to initialize the action.

For example, you want to fletch some regular logs, you use your knife on the logs and then press 1 to signal you want to fletch arrow shafts (as it's the first option in the item box)

Spacebar support has also been added, on which if you use key 2 for example, it'll remember that you pressed 2 and will act as though you pressed 2. This resets on a new item box being produced, and defaults to 1.

An example of a reset would be spacebar for fletching longbow (u) and then going to the spinning wheel, on opening the spinning wheel options, space bar will now act as key 1 and attempt to spin wool. This is also reset for fletching now, and spacebar will act as key 1 should you attempt to fletch again,

### CODE REVIEW

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/api/ext/QueueTaskExt.kt [hunk 1]
```diff
@@ -355,17 +360,51 @@ suspend fun QueueTask.produceItemBox(
...
-    val msg = requestReturnValue as? ResumePauseButtonMessage ?: return
-    val child = msg.component
+    val keyMsg = requestReturnValue as? KeyTypedMessage
+    val msg = requestReturnValue as? ResumePauseButtonMessage
+
+    var child = -1
+
+    // if the queue was returned with ResumePauseButton
+    if(msg != null) {
+        child = msg.component
```
Comment (Paleocene) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/api/ext/QueueTaskExt.kt:380]: This will be overwritten again if keyMsg != null, is that correct?

**Comment by alycii** [comment]
Code is complete, I've tested just about every edge-case I could think of. Would love someone to review and then test themselves :)

**Review by Paleocene — APPROVED** [review]
Some things I found, that may also be picked up in another PR:
- You have to press space bar each time you want to go to the next dialogue. Can't keep it pressed to go through dialogues
- There's no feedback when spacebarring dialogue. I.e., the "click here to continue" doesn't change to "please wait" or whatever it should be

Other than that: looks good!

**Comment by alycii** [comment]
> Some things I found, that may also be picked up in another PR:
> 
> - You have to press space bar each time you want to go to the next dialogue. Can't keep it pressed to go through dialogues
> 
> - There's no feedback when spacebarring dialogue. I.e., the "click here to continue" doesn't change to "please wait" or whatever it should be
> 
> 
> 
> Other than that: looks good!


Both of these issues will/could be addressed by sending the ResumePauseButton from the client via key events, but for the sake of simplicity, this is the temporary solution. 

In other words, I'm being lazy and refusing to start the client refactor and once I start doing some more of that, we'll do it the proper way 🤓
